File: Easy Guys/Sort Integers by The Number of 1 Bits/solution.py
class Solution:
    def sortByBits(self, arr: List[int]) -> List[int]:
        pq = []
        for num in arr:
            bits = 0
            for bit in range(14):
                if (num >> bit) & 1:
                    bits += 1
            heappush(pq, (bits, num))
        result = []
        while pq:
            result.append(heappop(pq)[1])
        return result

File: Easy Guys/Sort Integers by The Number of 1 Bits/solution.cpp
class Solution {
public:
    vector<int> sortByBits(vector<int>& arr) {
        sort(begin(arr), end(arr));
        vector<vector<int>> bits(14);
        for(int num: arr) {
            bits[popcount(static_cast<uint>(num))].push_back(num);
        }
        vector<int> result;
        for(auto &arr: bits) {
            for(auto num: arr) {
                result.push_back(num);
            }
        }
        return result;
    }
};
